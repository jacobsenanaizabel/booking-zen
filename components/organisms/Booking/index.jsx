import React, { useState, useEffect } from "react";
import { useSnackbar } from 'react-simple-snackbar';
import FullCalendar from "@fullcalendar/react";
import dayGridPlugin from "@fullcalendar/daygrid";
import timeGridPlugin from "@fullcalendar/timegrid";
import interactionPlugin from "@fullcalendar/interaction";
import dataEvents from "./mockedCalendar.json";


import {
	ContainerCalendar
  } from './styles';

const unavailable = "rgb(177, 160, 238)";

export const options = {
	position: 'top-center',
	style: {
	  backgroundColor: '#4a31a1',
	  color: '#ffffff',
	  fontSize: '20px',
	  textAlign: 'center',
	},
	closeStyle: {
	  color: '#ffffff',
	  fontSize: '16px',
	},
  };

const Booking = () => {
	let [convertedEvents, setConverted] = useState();
	const [openSnackbar] = useSnackbar(options)

	function handleDateClick() {
	}

	function handleEventClick(info) {
		toggleAvailabilityByProps(info);
	}

	function toggleAvailabilityByProps(info) {
		if (info.event.extendedProps.available) {
			info.el.style.backgroundColor = unavailable;
			info.el.style.borderColor = unavailable;
			const day = String(info.event.start.getDate()).padStart(2, "0");
			const month = String(info.event.start.getMonth()).padStart(2, "0");
			const year = String(info.event.start.getFullYear()).padStart(2, "0");
			const hour = String(info.event.start.getHours()).padStart(2, "0");
			const minutes = String(info.event.start.getMinutes()).padStart(2, "0");
			info.event.setExtendedProp("available", false);

			openSnackbar(`scheduled: ${day}/${month}/${year} ${hour}:${minutes}hrs`)
		} else {
			openSnackbar("Unavailable time");
		}
	}

	useEffect(() => {
		dataEvents.map((c) => {
			if (!c.available) {
				c.color = unavailable;
			}
		});

		setConverted(dataEvents);
	}, [dataEvents]);

	return (
		<>
			{convertedEvents && (
				<ContainerCalendar>
				<FullCalendar
					plugins={[dayGridPlugin, timeGridPlugin, interactionPlugin]}
					initialView="timeGridWeek"
					hiddenDays={[0, 6]}
					allDaySlot={false}
					height= "22rem"
					width="31rem"
					aspectRatio="3"
					expandRows={true}
					header={{
						center: 'title',
						right: 'prev,next'
					}}
					headerToolbar={{
						start: 'title', 
						end: 'prev,next', // will normally be on the right. if RTL, will be on the left
					  }}
					slotMinTime="08:00:00"
					slotMaxTime="18:00:00"
					events={convertedEvents}
					eventColor="#7655e9"
					dateClick={handleDateClick}
					eventClick={handleEventClick}
				/>
				</ContainerCalendar>
			)}
		</>
	);
};
export default Booking