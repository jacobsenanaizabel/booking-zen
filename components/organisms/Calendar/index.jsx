import React from 'react'
import FullCalendar from '@fullcalendar/react'
import timeGridPlugin from "@fullcalendar/timegrid";
import interactionPlugin from "@fullcalendar/interaction"

export default () => (
	<FullCalendar
	  plugins={[interactionPlugin, timeGridPlugin]}
	  initialView='timeGridWeek'
	  nowIndicator={true}
	  editable={true}
	  initialEvents={[
		{ title: 'nice event', start: new Date() }
	  ]}
	/>
  )