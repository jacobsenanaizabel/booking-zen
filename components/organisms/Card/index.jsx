import React from 'react';
import PropTypes from 'prop-types';
import { Avatar, Price, Description, Info } from '@components/atoms';
import  Rating  from '@components/atoms/Rating';

import {
  Container,
  About,
  Profile,
  TextStyled
} from './styles';

const Card = ({name, img, location, profession, price, rating, ratingNumbers, description}) => (
  <>
  <Container>
  <Profile>
    <Avatar img={img}/>
    <About>
        <TextStyled text={name} size="1.5rem" color="black"/>
        <Info location={location} profession={profession} />
        <Rating rating={rating} ratingNumbers={ratingNumbers}></Rating>
        <Price mount={price}/>
    </About>
  </Profile>
    <Description text={description}/>
  </Container>
  </>
);

Card.propTypes = {
  name: PropTypes.string,
  location: PropTypes.string,
  profession: PropTypes.string,
  price: PropTypes.number,
  rating: PropTypes.number,
  ratingNumbers: PropTypes.string,
  img: PropTypes.string
};

Card.defaultProps = {
  name: '',
  location: '',
  profession: '',
  price: 0,
  rating: 1,
  ratingNumbers: '0',
  img: '/'
};

export default Card;
