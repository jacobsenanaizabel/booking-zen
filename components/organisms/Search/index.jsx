import React, {useState } from 'react';
import { Button, Input } from '@components/atoms';

import {
  Container
} from './styles';

const Search = ({searchFreelancer}) => {

  const [search, setSearch] = useState();

  function onChangeSearching(evt){
    setSearch(evt.target.value)
  }

  const getFreelancers = (() =>{
    if(search !== undefined && search.length > 0){
      searchFreelancer(search)
    }
  })

  function onKeyDown(evt){
    if (evt.key === 'Enter') {
      searchFreelancer(evt.target.value)
    }
  }


  return(
    <>
        <Container>
          <Input
              name="search"
              onChange={onChangeSearching}
              onKeyDown={onKeyDown}
              placeholder="Find an expert by specialty or name" 
              type="text"
              value={search}
            />
          <Button text="Search" onClick={getFreelancers}/>
        </Container>
    </>
  )
};


export default Search;
