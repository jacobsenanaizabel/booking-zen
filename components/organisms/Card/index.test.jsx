import React from 'react';
import { render } from "@testing-library/react";
import {Card} from '@components/organisms/index';

describe('Card tests', () => {

  it('should match snapshot with default Card', () => {
    
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with default Card with searchFreelance', () => {
  
    const props = {
        img: 'http:img.com',
        name: 'test',
        location: 'now',
        rating: 5,
        ratingNumbers: '2 comments',
        price: 100,
        profession: 'coach'
    }
    const { asFragment } = render(<Card {...props}/>);
    expect(asFragment()).toMatchSnapshot();
  });

});