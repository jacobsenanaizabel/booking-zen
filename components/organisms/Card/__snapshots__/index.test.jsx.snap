// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should match snapshot with default Card 1`] = `
<DocumentFragment>
  <div
    class="styles__Container-sc-173llyf-0 eAfLlQ"
  >
    <div
      class="styles__Profile-sc-173llyf-2 gTWmfi"
    >
      <div
        class="styles__ContainerAvatar-sc-2niw0e-0"
      >
        <figure>
          <img
            class="styles__User-sc-2niw0e-1 eqmrgD"
            src="/"
          />
        </figure>
      </div>
      <div
        class="styles__About-sc-173llyf-1 bWkefv"
      >
        <p
          class="styles__Text-sc-tpjft1-0 byLudI"
          color="black"
        />
        <div
          class="styles__ContainerInfo-sc-ks18od-0"
        >
          <div
            class="styles__Info-sc-ks18od-1 gZmPJP"
          >
            <p
              class="styles__Text-sc-tpjft1-0 llQDMz"
              color="black"
            />
            <p
              class="styles__Text-sc-tpjft1-0 ghGUWi"
              color="purple"
            >
              - 
            </p>
          </div>
        </div>
        <div>
          Stars
        </div>
        <div
          class="styles__ContainerPrice-sc-l5bh20-0 dJJZvy"
        >
          <p
            class="styles__Text-sc-tpjft1-0 eFfUut"
            color="text"
          >
            0
          </p>
          <p
            class="styles__Text-sc-tpjft1-0 jGCNOp"
            color="text"
          >
             / 50 minutos
          </p>
        </div>
      </div>
    </div>
    <div
      class="styles__ContainerDescription-sc-1j7k9k3-0 eveMRK"
    >
      <p
        class="styles__Text-sc-tpjft1-0 jGCNOp"
        color="text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card tests should match snapshot with default Card with searchFreelance 1`] = `
<DocumentFragment>
  <div
    class="styles__Container-sc-173llyf-0 eAfLlQ"
  >
    <div
      class="styles__Profile-sc-173llyf-2 gTWmfi"
    >
      <div
        class="styles__ContainerAvatar-sc-2niw0e-0"
      >
        <figure>
          <img
            class="styles__User-sc-2niw0e-1 eqmrgD"
            src="http:img.com"
          />
        </figure>
      </div>
      <div
        class="styles__About-sc-173llyf-1 bWkefv"
      >
        <p
          class="styles__Text-sc-tpjft1-0 byLudI"
          color="black"
        >
          test
        </p>
        <div
          class="styles__ContainerInfo-sc-ks18od-0"
        >
          <div
            class="styles__Info-sc-ks18od-1 gZmPJP"
          >
            <p
              class="styles__Text-sc-tpjft1-0 llQDMz"
              color="black"
            >
              coach
            </p>
            <p
              class="styles__Text-sc-tpjft1-0 ghGUWi"
              color="purple"
            >
              - now
            </p>
          </div>
        </div>
        <div>
          Stars
        </div>
        <div
          class="styles__ContainerPrice-sc-l5bh20-0 dJJZvy"
        >
          <p
            class="styles__Text-sc-tpjft1-0 eFfUut"
            color="text"
          >
            100
          </p>
          <p
            class="styles__Text-sc-tpjft1-0 jGCNOp"
            color="text"
          >
             / 50 minutos
          </p>
        </div>
      </div>
    </div>
    <div
      class="styles__ContainerDescription-sc-1j7k9k3-0 eveMRK"
    >
      <p
        class="styles__Text-sc-tpjft1-0 jGCNOp"
        color="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
